torch/distributed/distributed_c10d.py

```diff
@@ -2819,6 +2819,12 @@ def all_gather_into_tensor(output_tensor, input_tensor, group=None, async_op=Fal
         _warn_not_in_group("all_gather_into_tensor")
         return
 
+    group = group or _get_default_group()
```

Comment:
Ok, here's my idea:

Add `torch/distributed/c10d_fallbacks.py` which expose functions with the same signature as c10d but with the additional contract of using fallbacks if necessary.

Could of things for us to discuss:

- should we inform the user that a fallback was used? if we should, how?
- should we allow users to control which devices are allowed to have a fallback. This is basically to guard against using fallbacks with NCCL.